devito/types/dimension.py

```diff
@@ -898,39 +901,52 @@ class ConditionalDimension(DerivedDimension):
     is_NonlinearDerived = True
     is_Conditional = True
 
-    __rkwargs__ = DerivedDimension.__rkwargs__ + ('factor', 'condition', 'indirect')
+    __rkwargs__ = DerivedDimension.__rkwargs__ + \
+        ('symbolic_factor', 'factor', 'condition', 'indirect')
 
     def __init_finalize__(self, name, parent=None, factor=None, condition=None,
-                          indirect=False, **kwargs):
+                          indirect=False, symbolic_factor=None, **kwargs):
         # `parent=None` degenerates to a ConditionalDimension outside of
         # any iteration space
         if parent is None:
             parent = BOTTOM
 
         super().__init_finalize__(name, parent)
 
-        # Always make the factor symbolic to allow overrides with different factor.
+        # Process subsampling factor
+        fname = f"{name}f"
         if factor is None or factor == 1:
             self._factor = None
-        elif is_integer(factor):
-            self._factor = Constant(name="%sf" % name, value=factor, dtype=np.int32)
-        elif factor.is_Constant and is_integer(factor.data):
-            self._factor = factor
+        elif is_number(factor):
+            self._factor = int(factor)
+        elif factor.is_Constant:
+            self._factor = factor.data
+            fname = factor.name
         else:
             raise ValueError("factor must be an integer or integer Constant")
 
+        if self._factor is not None:
+            # Always make the factor symbolic to allow overrides with different factor.
+            self._symbolic_factor = symbolic_factor or \
```

Comment:
why does this need to be an instance attribute? just like for example `symbolic_size` or `symbolic_max` are cached_property, can (should) this not be the same?

Reply:
Need it there so that it uses the name of the input `Constant` when one is used for backward compatibility. Can't do that in a cached_property.